### PR TITLE
Direct sampling 13.56Mhz on rtl-sdr

### PR DESCRIPTION
**Pull request by vinicentus**
Hi!

I've created a proof of concept for using direct sampling on the rtl-sdr device.
This means we can listen directly to the main carrier frequency of 13.56Mhz.

With my limited testing, and completely wrong antenna type, I still got great results.
Almost all pakcets are correctly read, as far as I can tell:
image

Now in order to include this in the upstream project, some things still need to be done.
I'm thinking:
1. This needs to be configurabe from the config file, instead of being hardcoded in [RealtekDevice.cpp#L71](https://github.com/vinicentus/nfc-laboratory/blob/direct-sampling/src/nfc-lib/lib-sdr/sdr-io/src/main/cpp/RealtekDevice.cpp#L71)
2. This change needs to be documented, at least in the readme.

So yeah. This is a work in progress, and therefore marked as a draft pull request.
But hopefully you find this useful, and hopefully it can be merged at some point!

Best regards,
Vincent

### CODE REVIEW

**Comment by josevcm** [comment]
Thank you very much for adding improvements to the project! I will integrate the changes as soon as possible and add the configuration from file.

**Comment by vinicentus** [comment]
Great!
I removed the draft flag from the PR now. So if you want you can merge. You should also have access to push commits to my direct-sampling branch automatically (seems to be a new github feature), if you want to work on this branch before merging.

I tired to make direct sampling configurable from the config file, but ran into an issue: Since only the rtl-sdr can do this (I think?), there needs to be a check that we aren't using some other sdr. I tried checking for the RealtekDevice subclass and casting to it when reading the config, but couldn't figure it out...

**Review by josevcm — APPROVED** [review]
Changes are correct

**Comment by josevcm** [comment]
Support for direct sampling activated by default, pending add configuration from file

**Comment by josevcm** [comment]
forget to hit merge!

**Comment by vinicentus** [comment]
Awesome!

Btw this is a super useful project, thanks for creating it!😄

**Comment by josevcm** [comment]
Please check an test recent commits to the master branch. I have included the configuration from the file but I cannot test it because my RTL-SDR does not support directSampling. When I activate it it does not receive anything :(

**Comment by vinicentus** [comment]
Ok, I should have time to try ut tomorrow

**Comment by vinicentus** [comment]
I had some time to test the new changes on master.
TLDR: it works as expected, but the documentation is slightly incaccurate.

When I run the project using qt creator, the default config seems to be this:
```[window]
timeFormat=false
followEnabled=true
filterEnabled=true

[device.rtlsdr]
centerFreq=13560000
sampleRate=10000000
gainMode=1
gainValue=439

[decoder.nfca]
enabled=true

[decoder.nfcb]
enabled=true

[decoder.nfcf]
enabled=true

[decoder.nfcv]
enabled=true
```

With this config, I don't get any usable results.

However by copy pasting the values from [dat/config/nfc-lab.conf](https://github.com/josevcm/nfc-laboratory/blob/master/dat/config/nfc-lab.conf) everything works.
The signal I pick up is weak and there are many decoding errors. But this is expected with my setup.

Direct sampling works when I change the config to
```
centerFreq=13560000
sampleRate=2400000
directSampling=2
```

About the documentation:
At least in my case, I get much better results when enabling direct sampling on the Q-branch (`directSampling=2`).
When using the I-branch (`directSampling=1`), the signal is too weak to decode.
So this should probably be reflected in the readme.

Maybe write something like:
You can use direct sampling on either the Q- or I-branch. The Q-branch is preferred due to better results.
You can use the I-branch with `directSampling=1` and the Q-branch with `directSampling=2`.
`directSampling=0` turns off direct sampling.

**Comment by josevcm** [comment]
Thanks Vincent, I will fix the documentation tomorrow.

**Comment by vinicentus** [comment]
I saw that you updated the documentation!

Time for a new release build? 😉😅

**Comment by josevcm** [comment]
Yes! next weekend I will prepare next release.
